aana/api/request_handler.py

```diff
@@ -229,3 +230,46 @@ async def _async_chat_completions(
                 "created": int(time.time()),
                 "choices": [{"index": 0, "message": response["message"]}],
             }
+
+    @app.get("/api/status", response_model=SDKStatusResponse)
+    async def status(self) -> SDKStatusResponse:
+        """The endpoint for checking the status of the application.
+
+        Returns:
+            SDKStatusResponse: The response containing the status of the application.
+        """
+        serve_status = serve.status()
+        message = ""
+        if any(
+            app.status == "DEPLOY_FAILED"
+            or app.status == "UNHEALTHY"
+            or app.status == "NOT_STARTED"
+            for app in serve_status.applications.values()
+        ):
+            sdk_status = SDKStatus.UNHEALTHY
+            error_messages = []
+            for app_name, app_status in serve_status.applications.items():
```

Comment:
Should we check for all the applications on the serve or just the current app?

Reply:
If yes, maybe we can return the response individually (as a dict) in the response!

Reply:
First, I want to clarify something. We have Aana apps and Ray Serve apps. They are different. One Aana app consists of multiple Ray Serve apps: one for request hander and one per each Aana deployment. You can run multiple Ray Serve apps and multiple Aana apps on the same Ray cluster. Although it is possible to run multiple Aana apps on the same Ray cluster, right now we mostly have one app per cluster.

The status endpoint should return the status of the current Aana app. Right now, we don't have a way to find which Ray Serve apps on the cluster belong to the current Aana app in the request handler but I will add that. 

So to answer your question, the status endpoint has to check all the Ray Serve apps that belong to it.

We can return detailed information for each Ray Serve app as well. Something like this:

```json
{
  "deployments": {
    "app": {
      "message": "",
      "status": "RUNNING"
    },
    "lowercase_deployment": {
      "message": "Error: Lowercase (lowercase_deployment): A replica's health check failed. This deployment will be UNHEALTHY until the replica recovers or a new deploy happens.",
      "status": "UNHEALTHY"
    },
    "task_queue_deployment": {
      "message": "",
      "status": "RUNNING"
    }
  },
  "message": "",
  "status": "UNHEALTHY"
}
```

Is that what you suggest?

Reply:
Added it to the PR. Please, review it again.

Reply:
Much better